Fable.Remoting.Middleware/FableMiddlewareAdapter.fs

```diff
@@ -0,0 +1,52 @@
+namespace Fable.Remoting.Middleware
+
+open Microsoft.AspNetCore.Builder
+open Microsoft.AspNetCore.Http
+
+open System
+open System.IO
+open Fable.Remoting.Server.SharedCE
+open FSharp.Control.Tasks
+
+[<AutoOpen>]
+module FableMiddlewareAdapter =
+
+  type BuilderOptions<'ctx> with
+    member this.WithBuilder(b) = { this with Builder = b }
+    member this.WithLogger(l) = { this with Logger = Some l }
+    member this.WithErrorHandler(eh) = {this with ErrorHandler = Some eh}
+
+  type RemotingOptions = BuilderOptions<HttpContext>
+
+  type FableRemotingMiddleware(next          : RequestDelegate,
+                               implementation: obj,
+                               options       : RemotingOptions) =
+
+      do if isNull next then raise (ArgumentNullException("next"))
```

Comment:
Just a little improvement: `raise (ArgumentNullException("next"))` could be written as `nullArg "next"`

Reply:
Done, thanks!